src/windshaft/layergroup-config.js

```diff
@@ -7,13 +7,14 @@ LayerGroupConfig.generate = function (options) {
   var layers = options.layers;
   var dataviews = options.dataviews;
   var config = { layers: [] };
-  _.each(layers, function (layer) {
+  _.each(layers, function (layer, index) {
+    var provisionalCartoCSS = options.styles[index]
     if (layer.isVisible()) {
       var layerConfig = {
         type: layer.get('type').toLowerCase(),
         options: {
           sql: layer.get('sql'),
-          cartocss: layer.get('cartocss'),
+          cartocss: provisionalCartoCSS || layer.get('initialStyle') || layer.get('cartocss'),
```

Comment:
I can see that, but I'm not sure why/when would we want to use the initial style of the layer... For instance, if `cartocss` is changed... why should `initialStyle` take precedence over `cartocss`?